text/0008-sync-and-storage-review-packet.md

```diff
@@ -0,0 +1,371 @@
+
+We think the quickest and lowest-risk approach is to intermediate some writes to Places, turning parts of the Places database into a read-only materialized view. This stress-tests the client-side data pipeline, and allows us to explore syncing new data (_e.g._, containers, alternative representations for keywords and tags), recovering Places data from the log, _etc_., without disturbing existing consumers.
+
+At this stage we hope to find and fix scaling and performance issues before doing further work. If we fail here, the process has been both informative and relatively cheap.
+
+As this phase proceeds we'll design and build production server capacity for the sync system, begin to address fit-and-finish concerns like push notification integration, and continue to assess requirements from new product efforts.
+
+### Phase 3: consolidate (2018Q3–)
+
+If we're confident in our new sync implementation, can use it from apps, and can handle Places-level scale, then we can proceed with consolidation and simplification to add value to Firefox. We can fan out: pick suitable storage (_e.g_., data that's high priority for syncing, an old store that's not owned by an existing team, …), make sure it has an async API, and port it on top of the log, following the example of Places.
+
+Early in this phase our documentation will go from adequate to great, we will place heavier emphasis on the developer experience and UI programming integration (_e.g._, GraphQL), and we expect to see further adoption in mobile apps and on the server as the cost of accessing Firefox data decreases.
+
+We can also assess the urgency of exposing this kind of storage — both access to Firefox's data, and dedicated storage — to WebExtensions.
+
+These additional efforts will likely be assessed and planned closer to the time; it would be unrealistic to estimate resourcing or timing when priorities are likely to shift.
+
+## Resources
+
+[Requirements capture for Sync](https://docs.google.com/document/d/1XOJhW-qf4iIyGHwiSLgWEfXYDWLy44So5L8CdwH6pcg)
+
+[Scoped encryption keys and access control for FxA](https://docs.google.com/document/d/1IvQJFEBFz0PnL4uVlIvt8fBS_IPwSK-avK0BRIHucxQ)
+
+Vision statements
+
+*   [Ryan Kelly](https://docs.google.com/document/d/1MrXNcVzDDYyGSvud6CvVSoPGNyvpJvHGSDpRwIcalQc)
+
+Blog posts
+
+*   [Thinking about Syncing, part 1](https://medium.com/@rnewman/thinking-about-syncing-part-1-timelines-7f758e2bd676) (2017)
+*   … [part 2](https://medium.com/@rnewman/thinking-about-syncing-part-2-timelines-and-change-90a0964f10f3)
+*   [Syncing and storage on three platforms](https://160.twinql.com/syncing-and-storage-on-three-platforms/) (2015)
+*   [Different kinds of storage](https://160.twinql.com/different-kinds-of-storage/) (user agent service) (2016)
+
+## Thanks
+
+Thanks to our readers: myk, asuth, mak, grisha, tspurway, snorp, overholt, ckarlof, wennie, selena, gps, trink, hildjj, and plenty more.
```

Comment:
s/readers/reviewers ?

Reply:
I was trying to do two things: firstly, to avoid colliding with the Reviewer role; secondly, hitting this definition of "reader":

> a person who reads and reports to a publisher or producer on the merits of manuscripts submitted for publication or production, or who provides critical comments on the text prior to publication

which is a little distinct from and seems a better fit than "reviewer":

> a person who formally assesses or examines something with a view to changing it if necessary